spec/features/take_one_time_shift.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe "Volunteer takes One-Time Shifts" do
+  let(:log) { create(:log, when: "Tue, 31 Jan 2017", donor: donor, recipients: [recipient]) }
+  let(:donor) { create(:donor, name: "This is a name") }
+  let(:volunteer) { create(:volunteer, :not_waived, regions: [log.region], assigned: true) }
+  let(:location) { create(:location) }
+  let(:recipient) { create(:recipient)}
+  # Expect page to show log entry
+  feature "An available one-time shift" do
+    before do
+      login volunteer
+      location
+      recipient
```

Comment:
Instead of calling these `let`s in the `before` action, we can use `let!`. Something like:

```diff
diff --git a/spec/features/take_one_time_shift.rb b/spec/features/take_one_time_shift.rb
index 57ca029..1caffb0 100644
--- a/spec/features/take_one_time_shift.rb
+++ b/spec/features/take_one_time_shift.rb
@@ -4,14 +4,12 @@ RSpec.describe "Volunteer takes One-Time Shifts" do
   let(:log) { create(:log, when: "Tue, 31 Jan 2017", donor: donor, recipients: [recipient]) }
   let(:donor) { create(:donor, name: "This is a name") }
   let(:volunteer) { create(:volunteer, :not_waived, regions: [log.region], assigned: true) }
-  let(:location) { create(:location) }
-  let(:recipient) { create(:recipient)}
+  let!(:location) { create(:location) }
+  let!(:recipient) { create(:recipient)}
   # Expect page to show log entry
   feature "An available one-time shift" do
     before do
       login volunteer
-      location
-      recipient
     end
 
     it "can view an available shift" do
```

More info: https://relishapp.com/rspec/rspec-core/v/3-4/docs/helper-methods/let-and-let